packages/tailwindcss/src/index.test.ts

```diff
@@ -4657,11 +4720,347 @@ describe('@variant', () => {
+  })
+
+  it('works with multiple `color-mix(…)` functions in one declaration', async () => {
+    await expect(
+      compileCss(
+        css`
+          @theme {
+            --color-red-500: oklch(63.7% 0.237 25.331);
+            --color-orange-500: oklch(70.5% 0.213 47.604);
+          }
+          @tailwind utilities;
+          .gradient {
+            background: linear-gradient(
+              90deg,
+              color-mix(in oklab, var(--color-red-500) 50%, transparent) 0%,
+              color-mix(in oklab, var(--color-orange-500) 50%, transparent) 0%,
+              100%
+            );
+          }
+        `,
+        [],
+      ),
+    ).resolves.toMatchInlineSnapshot(`
+      ":root, :host {
+        --color-red-500: oklch(63.7% .237 25.331);
+        --color-orange-500: oklch(70.5% .213 47.604);
+      }
+
+      .gradient {
+        background: linear-gradient(90deg, #fb2c3680 0%, #fe6e0080 0%, 100%);
+      }
+
+      @supports (color: color-mix(in lab, red, red)) {
+        .gradient {
+          background: linear-gradient(90deg, color-mix(in oklab, var(--color-red-500) 50%, transparent) 0%, color-mix(in oklab, var(--color-orange-500) 50%, transparent) 0%, 100%);
+        }
+      }"
+    `)
+  })
+
+  it('works with no spaces after the `var(…)`', async () => {
+    await expect(
+      compileCss(
+        // prettier-ignore
+        css`
+          @theme {
+            --color-red-500: oklch(63.7% 0.237 25.331);
+          }
+          @tailwind utilities;
+          .text-red-500\/50 {
+            color: color-mix(in oklab,var(--color-red-500)50%,transparent);
+          }
+        `,
+        [],
+      ),
+    ).resolves.toMatchInlineSnapshot(`
+      ":root, :host {
+        --color-red-500: oklch(63.7% .237 25.331);
+      }
+
+      .text-red-500\\/50 {
+        color: #fb2c3680;
+      }
+
+      @supports (color: color-mix(in lab, red, red)) {
+        .text-red-500\\/50 {
+          color: color-mix(in oklab, var(--color-red-500) 50%, transparent);
+        }
+      }"
+    `)
+  })
+
+  it('does not replace `currentColor` inside `color-mix(…)`', async () => {
+    await expect(
+      compileCss(
+        css`
+          @tailwind utilities;
+        `,
+        ['text-current/50'],
+      ),
+    ).resolves.toMatchInlineSnapshot(`
+      ".text-current\\/50 {
+        color: color-mix(in oklab, currentColor 50%, transparent);
+      }"
+    `)
+  })
+
+  it('also replaces eventual variables in opacity values', async () => {
+    await expect(
+      compileCss(
+        css`
+          @theme {
+            --my-half: 50%;
+            --color-red-500: oklch(63.7% 0.237 25.331);
+          }
+          @tailwind utilities;
+        `,
+        ['text-red-500/(--my-half)'],
+      ),
+    ).resolves.toMatchInlineSnapshot(`
+      ":root, :host {
+        --my-half: 50%;
+        --color-red-500: oklch(63.7% .237 25.331);
+      }
+
+      .text-red-500\\/\\(--my-half\\) {
+        color: #fb2c3680;
+      }
```

Comment:
Sad that this means that you can't change the `--my-half` value at runtime, but if it's not supported what can you do haha.